src/extensions/default/NavigationAndHistory/main.js

```diff
@@ -0,0 +1,402 @@
+/*
+ * Copyright (c) 2012 Adobe Systems Incorporated. All rights reserved.
```

Comment:
Just don't feel myself comfortable with it maybe because it is the first time I see it. @larz0 or @GarthDB may give an opinion on this design change.